specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/dataConnectorDefinitions.json

```diff
@@ -0,0 +1,706 @@
+      "required": [
+        "baseQuery",
+        "legend",
+        "metricName"
+      ],
+      "type": "object",
+      "properties": {
+        "metricName": {
+          "description": "Gets or sets the metric name that the query is checking. For example: 'Total data receive'.",
+          "type": "string"
+        },
+        "legend": {
+          "description": "Gets or sets the legend for the graph.",
+          "type": "string"
+        },
+        "baseQuery": {
+          "description": "Gets or sets the base query for the graph.\r\nThe base query is wrapped by Sentinel UI infra with a KQL query, that measures the volume over time.",
+          "type": "string"
+        }
+      }
+    },
+    "InstructionStep": {
+      "description": "Instruction steps to enable the connector.",
+      "type": "object",
+      "properties": {
+        "title": {
+          "description": "Gets or sets the instruction step title.",
+          "type": "string"
+        },
+        "description": {
+          "description": "Gets or sets the instruction step description.",
+          "type": "string"
+        },
+        "instructions": {
+          "description": "Gets or sets the instruction step details.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/InstructionStepDetails"
+          },
+          "x-ms-identifiers": []
+        },
+        "innerSteps": {
+          "description": "Gets or sets the inner instruction steps details.\r\nFoe Example: instruction step 1 might contain inner instruction steps: [instruction step 1.1, instruction step 1.2].",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/InstructionStep"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "InstructionStepDetails": {
+      "description": "Instruction step details, to be displayed in the Instructions steps section in the connector's page in Sentinel Portal.",
+      "required": [
+        "parameters",
+        "type"
+      ],
+      "type": "object",
+      "properties": {
+        "parameters": {
+          "description": "Gets or sets the instruction type parameters settings.",
+          "type": "object"
+        },
+        "type": {
+          "description": "Gets or sets the instruction type name.",
+          "type": "string"
+        }
+      }
+    },
+    "ConnectorDefinitionsPermissions": {
+      "description": "The required Permissions for the connector.",
+      "type": "object",
+      "properties": {
+        "tenant": {
+          "description": "Gets or sets the required tenant permissions for the connector.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "licenses": {
+          "description": "Gets or sets the required licenses for the user to create connections.",
+          "type": "array",
+          "items": {
+            "type": "string"
```

Comment:
>"type": "string" [](http://example.com/codeflow?start=12&length=16)

What string format does a "license" have? Beef up the description so a client knows what these should be.